p2p/src/sessions/mod.rs

```diff
@@ -117,8 +120,9 @@ where
             .set_limit(outbound_consolidated_limit);
     }
     /// Method to set the handshake timeout
-    pub fn set_handshake_timeout(&mut self, handshake_timeout: Duration) {
+    pub fn set_handshake_options(&mut self, handshake_timeout: Duration, ts_diff: i64) {
```

Comment:
I would prefer to have two different setters instead: `set_handshake_timeout` and `set_handshake_max_ts_diff`